### PR TITLE
PERF-#4866: `iloc` function that used in `partition.mask` should be serialized only once

### LINKED ISSUE #4866: PERF: `iloc` function that used in `partition.mask` should be serialized only once
Place: https://github.com/modin-project/modin/blob/cfcdf081e197d35e1fd1526f5585e361793dc2fa/modin/core/dataframe/pandas/partitioning/partition.py#L188

Motivation: `partition.mask` - one of the most used functions.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4866 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by jbrockmendel — DISMISSED** [review]
LGTM

**Review by mvashishtha — DISMISSED** [review]
@anmyachev LGTM, but do you have any idea how useful this is for performance?

**Comment by anmyachev** [comment]
> @anmyachev LGTM, but do you have any idea how useful this is for performance?

I did not specifically measure, but I think the acceleration is of the same order as in https://github.com/modin-project/modin/pull/4861#issuecomment-1227243308

**Review by pyrito — DISMISSED** [review]

File: modin/core/dataframe/pandas/partitioning/partition.py [hunk 1]
```diff
@@ -162,6 +162,12 @@ def to_numpy(self, **kwargs):
         """
         return self.apply(lambda df, **kwargs: df.to_numpy(**kwargs)).get()
 
+    _iloc = staticmethod(
+        lambda df, row_labels, col_labels: df.iloc[  # noqa: E731 (lambda assignment)
+            row_labels, col_labels
+        ]
+    )
+
```
Comment (vnlitvinov) [inline modin/core/dataframe/pandas/partitioning/partition.py:170]:
```suggestion
    @staticmethod
    def _iloc(df, row_labels, col_labels):
        return df.iloc[row_labels, col_labels]
```
looks shorter and cleaner 😉

File: modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py [hunk 2]
```diff
@@ -206,6 +206,8 @@ def __copy__(self):
             call_queue=self.call_queue,
         )
 
+    _iloc = ray.put(lambda df, row_labels, col_labels: df.iloc[row_labels, col_labels])
```
Comment (vnlitvinov) [inline modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py:209]:
```suggestion
    _iloc = ray.put(PandasDataframePartition._iloc)
```
to not duplicate the implementation

**Review by YarShev — APPROVED** [review]

**Pull request merged**